source/module_hamilt_lcao/module_gint/grid_technique.cpp

```diff
@@ -26,17 +26,60 @@ Grid_Technique::Grid_Technique()
 
 Grid_Technique::~Grid_Technique()
 {
-    delete[] nlocdimg;
-    delete[] nlocstartg;
-    delete[] nad;
-    delete[] how_many_atoms;
-	delete[] start_ind;
-	delete[] which_atom;
-	delete[] which_bigcell;
-	delete[] which_unitcell;
-	delete[] bcell_start;
-	delete[] in_this_processor;
-	delete[] trace_lo;
+    if(nlocdimg!=nullptr)
+	{
+		delete[] nlocdimg;
+	}
```

Comment:
delete a nullptr is OK; there is no need to determine if it is a nullptr before deleting it.